### PR TITLE
[libgxps] Remove folder bin at static build

### PR DESCRIPTION
**Pull request by LilyWangLL**
`libgxps` only builds shared lib, so add `vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)` for fix this failure from [baseline](https://dev.azure.com/vcpkg/public/_build/results?buildId=106144&view=logs&j=3859ca13-59fe-57bd-13f9-449df7d191a6&t=128b3fda-02b4-5a50-e033-dfd6a334bd4e). 
The following codes from the upstream: https://gitlab.gnome.org/GNOME/libgxps/-/blob/master/libgxps/meson.build?ref_type=heads#L76.
```
gxps = shared_library('gxps',
                      include_directories: core_inc,
                      sources: sources + headers + private_headers + [ gxps_version_h ],
                      soversion: soversion,
                      version: libversion,
                      install: true,
                      dependencies: gxps_deps,
                      c_args: extra_args + common_flags + [
                        '-DG_LOG_DOMAIN="GXPS"',
                        '-DGXPS_COMPILATION',
                      ],
                      link_args: common_ldflags)
```

```
-- Performing post-build validation
D:\a\_work\1\s\ports\libgxps\portfile.cmake: warning: DLLs should not be present in a static build, but the following DLLs were found. To suppress this message, add set(VCPKG_POLICY_DLLS_IN_STATIC_LIBRARY enabled)
D:\p\libgxps_x64-windows-static: note: the DLLs are relative to ${CURRENT_PACKAGES_DIR} here
note: debug/bin/gxps-2.dll
note: bin/gxps-2.dll
D:\a\_work\1\s\ports\libgxps\portfile.cmake: warning: ${CURRENT_PACKAGES_DIR}/debug/bin exists but should not in a static build. To suppress this message, add set(VCPKG_POLICY_DLLS_IN_STATIC_LIBRARY enabled)
D:\a\_work\1\s\ports\libgxps\portfile.cmake: warning: ${CURRENT_PACKAGES_DIR}/bin exists but should not in a static build. To suppress this message, add set(VCPKG_POLICY_DLLS_IN_STATIC_LIBRARY enabled)
note: if creation of these directories cannot be disabled, you can add the following in portfile.cmake to remove them
if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
  file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/bin" "${CURRENT_PACKAGES_DIR}/bin")
endif()
D:\a\_work\1\s\ports\libgxps\portfile.cmake: warning: Found 3 post-build check problem(s). These are usually caused by bugs in portfile.cmake or the upstream build system. Please correct these before submitting this port to the curated registry.
error: building libgxps:x64-windows-static failed with: POST_BUILD_CHECKS_FAILED
```
The related issue: #40527.
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [ ] ~~SHA512s are updated for each updated download.~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version.~~
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by dg0yt** [comment]
Not the proper solution.
BTW openscap has installation order bugs since the update.

**Comment by dg0yt** [comment]
Openscap now handled in #40575.

File: ports/libgxps/vcpkg.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   "port-version": 4,
   "description": "libgxps is a GObject based library for handling and rendering XPS documents.",
   "homepage": "https://wiki.gnome.org/Projects/libgxps",
+  "supports": "!staticcrt",
```
Comment (dg0yt) [inline ports/libgxps/vcpkg.json:7]: Cf. https://github.com/microsoft/vcpkg/pull/40575#discussion_r1727720244.

Comment (LilyWangLL) [reply ports/libgxps/vcpkg.json]: Thanks for your comment, I remove this `supports` and use add `libgxps:x64-windows-static=fail` instead.

**Review by Cheney-W — APPROVED** [review]

**Pull request merged**